src/panels/config/backup/ha-config-backup-dashboard.ts

```diff
@@ -1,171 +1,278 @@
+} from "../../../components/data-table/ha-data-table";
 import "../../../components/ha-button";
 import "../../../components/ha-card";
+import "../../../components/ha-circular-progress";
+import "../../../components/ha-fab";
 import "../../../components/ha-icon";
 import "../../../components/ha-icon-next";
-import "../../../components/ha-list-item";
-import "../../../layouts/hass-subpage";
-
-import { navigate } from "../../../common/navigate";
-import { fetchBackupAgentsInfo } from "../../../data/backup";
+import "../../../components/ha-summary-card";
+import "../../../components/ha-svg-icon";
+import {
+  fetchBackupAgentsBackups,
+  fetchBackupInfo,
+  generateBackup,
+  type BackupContent,
+} from "../../../data/backup";
+import "../../../layouts/hass-tabs-subpage-data-table";
 import { SubscribeMixin } from "../../../mixins/subscribe-mixin";
-import type { HomeAssistant } from "../../../types";
+import type { HomeAssistant, Route } from "../../../types";
 import { brandsUrl } from "../../../util/brands-url";
+import {
+  showAlertDialog,
+  showConfirmationDialog,
+} from "../../lovelace/custom-card-helpers";
 
 @customElement("ha-config-backup-dashboard")
 class HaConfigBackupDashboard extends SubscribeMixin(LitElement) {
   @property({ attribute: false }) public hass!: HomeAssistant;
 
   @property({ type: Boolean }) public narrow = false;
 
-  @state() private _agents: { id: string }[] = [];
+  @property({ attribute: false }) public route!: Route;
 
-  protected firstUpdated(changedProps) {
-    super.firstUpdated(changedProps);
-    this._fetchAgents();
-  }
+  @state() private _backingUp = false;
+
+  @state() private _backups: BackupContent[] = [];
+
+  private _columns = memoizeOne(
+    (
+      narrow,
+      _language,
+      localize: LocalizeFunc
+    ): DataTableColumnContainer<BackupContent> => ({
+      name: {
+        title: localize("ui.panel.config.backup.name"),
+        main: true,
+        sortable: true,
+        filterable: true,
+        flex: 2,
+        template: (backup) =>
+          narrow || !backup.path
+            ? backup.name
+            : html`${backup.name}
+                <div class="secondary">${backup.path}</div>`,
+      },
+      path: {
+        title: localize("ui.panel.config.backup.path"),
+        hidden: !narrow,
+        template: (backup) => backup.path || "-",
+      },
+      size: {
+        title: localize("ui.panel.config.backup.size"),
+        filterable: true,
+        sortable: true,
+        template: (backup) => Math.ceil(backup.size * 10) / 10 + " MB",
+      },
+      date: {
+        title: localize("ui.panel.config.backup.created"),
+        direction: "desc",
+        filterable: true,
+        sortable: true,
+        template: (backup) =>
+          relativeTime(new Date(backup.date), this.hass.locale),
+      },
+      locations: {
+        title: "Locations",
+        template: (backup) =>
+          html`${(backup.agents || []).sort().map((agent) => {
```

Comment:
Can we do the sorting and creation of array in the data function instead of render

Reply:
The API will evolve so this logic may change. I will change that once the api is ready (another PR)